webapp/channels/src/components/user_settings/general/user_settings_general.tsx

```diff
@@ -144,17 +151,29 @@ type State = {
     clientError?: string | null;
     serverError?: string | {server_error_id: string; message: string};
     emailError?: string;
+    customAttributeValues: Record<string, string>;
 }
 
 export class UserSettingsGeneralTab extends PureComponent<Props, State> {
     public submitActive = false;
 
     constructor(props: Props) {
         super(props);
-
         this.state = this.setupInitialState(props);
     }
 
+    componentDidMount() {
+        if (this.props.enableCustomProfileAttributes) {
+            const fetchValues = async () => {
```

Comment:
Should we use the `getCustomProfileAttributeValues` from `webapp/channels/src/packages/mattermost-redux/src/actions/general.ts`?

Reply:
We'll create followup tickets for these ones